### PR TITLE
workflow_job_template and workflow_job_template_node

### PR DESCRIPTION
**Pull request by jameswnl**
https://bugzilla.redhat.com/show_bug.cgi?id=1468339

Part of https://github.com/ManageIQ/manageiq/issues/17441

### CODE REVIEW

**Comment by jameswnl** [comment]
@bdunne can you take a look first? (specs are still being worked on)

**Comment by jameswnl** [comment]
@miq-bot remove_label wip
@miq-bot add_label enhancement

**Comment by jameswnl** [comment]
@miq-bot assign @bdunne

**Comment by jameswnl** [comment]
@miq-bot add_reviewer @bdunne

File: spec/collection_spec.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@
     AnsibleTowerClient::Job,
     AnsibleTowerClient::JobEvent,
     AnsibleTowerClient::JobTemplate,
+    AnsibleTowerClient::WorkflowJob,
```
Comment (bdunne) [inline spec/collection_spec.rb]: Can these be alphabetical?

Comment (bdunne) [reply spec/collection_spec.rb]: Shouldn't `WorkflowJobTemplateNode` be listed here too?

File: spec/support/mock_api.rb [hunk 2]
```diff
@@ -32,6 +32,10 @@ def get(path, get_options = nil)
         wrap_response(Job.response)
       when "job_templates"
         wrap_response(JobTemplate.response)
+      when "workflow_job_template_nodes"
```
Comment (bdunne) [inline spec/support/mock_api.rb]: Also, can these be alphabetical?

File: spec/support/mock_api/workflow_job_template_node.rb [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+          "url": "/api/v1/workflow_job_template_nodes/1/",
+          "related": {
+          "unified_job_template": "/api/v1/job_templates/216/",
+          "success_nodes": "/api/v1/workflow_job_template_nodes/1/success_nodes/",
+          "failure_nodes": "/api/v1/workflow_job_template_nodes/1/failure_nodes/",
+          "always_nodes": "/api/v1/workflow_job_template_nodes/1/always_nodes/",
+          "workflow_job_template": "/api/v1/workflow_job_templates/298/"
+        },
+          "summary_fields": {
+          "workflow_job_template": {
```
Comment (bdunne) [inline spec/support/mock_api/workflow_job_template_node.rb]: It looks like there's some indentation issue in this file

**Pull request merged**